packages/editor/src/plugins/interactive-video/toolbar.tsx

```diff
@@ -0,0 +1,56 @@
+import { PluginToolbar, PreviewButton } from '@editor/editor-ui/plugin-toolbar'
+import { DropdownButton } from '@editor/editor-ui/plugin-toolbar/plugin-tool-menu/dropdown-button'
+import { PluginDefaultTools } from '@editor/editor-ui/plugin-toolbar/plugin-tool-menu/plugin-default-tools'
+import { useEditStrings } from '@editor/i18n/edit-strings-provider'
+import { runChangeDocumentSaga } from '@editor/store'
+import { EditorPluginType } from '@editor/types/editor-plugin-type'
+import { faArrowsRotate } from '@fortawesome/free-solid-svg-icons'
+import { type Dispatch, type SetStateAction } from 'react'
+import { useDispatch } from 'react-redux'
+
+import type { InteractiveVideoProps } from '.'
+
+export const InteractiveVideoToolbar = ({
+  id,
+  state,
+  previewActive,
+  setPreviewActive,
+}: InteractiveVideoProps & {
+  previewActive: boolean
+  setPreviewActive: Dispatch<SetStateAction<boolean>>
+}) => {
+  const interactiveVideoStrings = useEditStrings().plugins.interactiveVideo
+  const dispatch = useDispatch()
+
+  function handleOnChangeVideo() {
+    dispatch(
+      runChangeDocumentSaga({
+        id: state.video.id,
+        state: { initial: (curr) => ({ ...(curr as object), src: '' }) },
```

Comment:
This way, the already set marks will stay on the new video. Not sure if that's desired, it feels like extra work for the user to remove the marks then. I'd prefer a warning explaining that the marks will be lost if the video is replaced, and then reset the whole state.

Reply:
Yeah, I considered this as well. This would definitely simplify things and we don't end up with marks that are actually after the timeline of the video etc.

I thought about use cases for replacing the video and thought people might upload a slightly changed version (like with louder sound, or without a backing track) and then keeping the marks would be great.

But to be honest we don't know enough about the way people will actually use it to say if that is common.
Especially since you can't move marks right now the utility is definitely pretty limited.

I will send this and the cancel-feedback to Gregor for a third opinion.

Reply:
Hey @hejtful and @elbotho ! I don't really know if this is thr right spot to reply to your UX issues:
1. Issue: Canceling a exercise creation process: In my opinion there are some ways to manage this problem. Firstly, we can shift the title field more to the center of the plugin and make a close button with TT: Schließen und Aufgabe löschen (Close and delete exercise) the _close-button_ would have a hover background (either red or serlo blue, 50% Opacity)
![Screenshot 2024-11-19 at 09 43 19](https://github.com/user-attachments/assets/145ca5f6-2437-48e6-ae7f-5f62a2cdd32b)
![Screenshot 2024-11-19 at 09 42 53](https://github.com/user-attachments/assets/47167354-c709-4491-8cda-33ba5efe5870)
I left the the close button unaligned on purpose and in a lighter grey (757575) to signalise the "outlying" function and make it less obvious and cause less visual friction.
I also increased the size of the Save button and aligned it centered with the title field to underline its importance.

Reply:
the other option would be indeed just signaling with a toast message, which I also would be fine with tbh. 
@elbotho do you have a preference?

Reply:
2. Issue: 
3 Options: 
1. Keep all of your exercises at the same time stamp and, if necessary delete them manually or with an option in the "..." Menu to delete all exercises with one click (with an Confirmation)
2. Automatically delete all exercises, but make it optional: When switching the video a quick popup: "Do you want to keep the existing exercises?"
3. (Out of scope imo) Keep all of the exercises and in addition implement a way to manually adjust the timestamp of the exercise.

For now I tend to option 1

Reply:
Second Issue, Option 1 implemented here:
https://github.com/serlo/frontend/pull/4320